<?xml version="1.0"?>
<launch>

    <arg name="namespace" default="bebop" />

    <remap from="fiducial_marker" to="/ar_pose_marker"/>
    <remap from="camera" to="$(arg namespace)/image_raw"/>
    <remap from="actions" to="iisc_bebop_nav/cmd"/>
    <remap from="output_image" to="iisc_autonomous_nav/output_image"/>

    <include file="$(find ar_track_alvar)/launch/bebop2_A1.launch"/>

    <node name="iisc_landing" pkg="iisc_autonav_outdoors" type="landing.py" output="screen">      
        <remap from="green" to="/iisc_landing/pad_detection/green_mask"/>
        <remap from="red" to="/iisc_landing/pad_detection/red_mask"/>
        <remap from="blue" to="/iisc_landing/pad_detection/blue_mask"/>
	    <rosparam command="load" file="$(find iisc_autonav_outdoors)/config/landing.yaml" />
    </node>

    <node name="iisc_junc_turning" pkg="iisc_autonav_outdoors" type="junc_turning.py" output="screen">
	    <remap from="odom" to="$(arg namespace)/odom"/>
        <rosparam command="load" file="$(find iisc_autonav_outdoors)/config/junction_calib.yaml" />
    </node>

    <node name="iisc_bebop_nav" pkg="iisc_autonav_outdoors" type="bebop_nav.py" output="screen">
        <remap from="cmd" to="/$(arg namespace)/cmd_vel"/>
        <remap from="land" to="$(arg namespace)/land"/>
        <remap from="takeoff" to="$(arg namespace)/takeoff"/>
        <remap from="cam_control" to="$(arg namespace)/camera_control"/>
        <remap from="battery" to="$(arg namespace)/states/common/CommonState/BatteryStateChanged"/>
        <rosparam command="load" file="$(find iisc_autonav_outdoors)/config/bebop_nav_vals.yaml" />
    </node>

    <node name="iisc_rf" pkg="iisc_autonav_outdoors" type="rf.py" output="screen">
        <remap from="logits" to="/iisc_rf/DNN_logits"/>
        <rosparam command="load" file="$(find iisc_autonav_outdoors)/config/rf_settings.yaml" />
    </node>

    <rosparam command="load" file="$(find iisc_autonav_outdoors)/config/iisc_autonomous_nav.yaml" />

</launch>